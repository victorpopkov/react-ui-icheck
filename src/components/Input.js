import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const iCheck = 'iCheck';
const iCheckHelper = `${iCheck}-helper`;

class Input extends Component {
  static propTypes = {
    inputType: PropTypes.string.isRequired,
    activeClass: PropTypes.string,
    activeClassName: PropTypes.string,
    aria: PropTypes.bool,
    checkboxClass: PropTypes.string,
    checked: PropTypes.bool,
    checkedClass: PropTypes.string,
    className: PropTypes.string,
    defaultChecked: PropTypes.bool,
    disabled: PropTypes.bool,
    disabledClass: PropTypes.string,
    focusClass: PropTypes.string,
    hoverClass: PropTypes.string,
    hovered: PropTypes.bool,
    id: PropTypes.string,
    increaseArea: PropTypes.string,
    indeterminate: PropTypes.bool,
    indeterminateClass: PropTypes.string,
    inheritClass: PropTypes.bool,
    inheritID: PropTypes.bool,
    insert: PropTypes.node,
    isMobile: PropTypes.bool,
    label: PropTypes.string,
    name: PropTypes.string,
    radioClass: PropTypes.string,
    value: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
  };

  static defaultProps = {
    activeClass: null,
    activeClassName: null,
    aria: false,
    checkboxClass: 'icheckbox',
    checked: false,
    checkedClass: 'checked',
    className: null,
    defaultChecked: false,
    disabled: false,
    disabledClass: 'disabled',
    focusClass: 'focus',
    hoverClass: 'hover',
    hovered: false,
    id: null,
    increaseArea: '',
    indeterminate: false,
    indeterminateClass: 'indeterminate',
    inheritClass: false,
    inheritID: false,
    insert: null,
    isMobile: false,
    label: null,
    name: null,
    radioClass: 'iradio',
    value: '',
    onBlur: null,
    onChange: null,
    onFocus: null,
  };

  constructor(props) {
    super(props);

    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleHelperClick = this.handleHelperClick.bind(this);

    const {
      checked,
      defaultChecked,
      hovered,
      indeterminate,
    } = props;

    this.state = {
      active: false,
      checked: defaultChecked || checked,
      focused: false,
      hovered,
      indeterminate,
    };
  }

  componentWillReceiveProps(nextProps) {
    const { checked: prevChecked, hovered: prevHovered } = this.props;
    const { checked, hovered } = nextProps;

    if (checked !== prevChecked) {
      this.setState({
        checked,
      });
    }

    if (hovered !== prevHovered) {
      this.setState({
        hovered,
      });
    }
  }

  handleBlur(event) {
    const { onBlur } = this.props;

    this.setState({
      focused: false,
    });

    if (onBlur) {
      onBlur(event);
    }
  }

  handleChange(event) {
    const { onChange } = this.props;
    const { checked } = event.target;

    this.setState(state => ({
      checked: !state.checked,
    }));

    if (onChange) {
      onChange(event, checked);
    }
  }

  handleFocus(event) {
    const { onFocus } = this.props;

    this.setState({
      focused: true,
    });

    if (onFocus) {
      onFocus(event);
    }
  }

  handleHelperClick(event) {
    const {
      disabled,
      inputType,
      insert,
      label,
      onChange,
    } = this.props;
    const { checked } = this.state;

    if ((!label && !insert) || disabled) {
      return;
    }

    if (inputType === 'checkbox') {
      this.setState(state => ({
        checked: !state.checked,
      }));
    }

    event.preventDefault();
    event.stopPropagation();

    if (onChange) {
      // make sure <ins /> element is not target
      event.target = this.checkbox; // eslint-disable-line no-param-reassign
      onChange(event, checked);
    }
  }

  render() {
    const {
      activeClass,
      activeClassName,
      aria,
      checkboxClass,
      checkedClass,
      className,
      disabled,
      disabledClass,
      focusClass,
      hoverClass,
      id,
      increaseArea,
      indeterminateClass,
      inheritClass,
      inheritID,
      inputType,
      insert,
      isMobile,
      name,
      radioClass,
      value,

      defaultChecked,
      hovered,
      indeterminate,

      ...other
    } = this.props;

    const {
      active,
      checked,
      focused,
      hovered: stateHovered,
      indeterminate: stateIndeterminate,
    } = this.state;

    const wrapProps = {
      className: classNames({
        [checkboxClass]: inputType === 'checkbox',
        [radioClass]: inputType === 'radio',
        [indeterminateClass]: stateIndeterminate,
        [checkedClass]: checked,
        [disabledClass]: disabled,
        [hoverClass]: stateHovered,
        [focusClass]: focused,
        [activeClassName || activeClass]: active,
      }),
    };

    if (inheritClass) {
      wrapProps.className = classNames(wrapProps.className, className);
    }

    if (inheritID && id) {
      wrapProps.id = `${iCheck}-${id}`;
    }

    if (aria) {
      wrapProps.role = inputType;
      // Set ARIA "labelledby"
      wrapProps['aria-labelledby'] = wrapProps.id;
      wrapProps['aria-checked'] = checked;
    }

    // Clickable area
    let area = (`${increaseArea}`).replace('%', '') || 0;
    if (area < -50) {
      area = -50;
    }

    // Layer styles
    const offset = `${-area}%`;
    const size = `${100 + (area * 2)}%`;
    const layer = {
      position: 'absolute',
      top: offset,
      left: offset,
      display: 'block',
      width: size,
      height: size,
      margin: 0,
      padding: 0,
      background: '#fff',
      border: 0,
      opacity: 0,
      cursor: disabled ? 'default' : 'pointer',
    };

    // Layer addition
    const helper = (
      <ins
        className={iCheckHelper}
        role="presentation"
        style={layer}
        onClick={this.handleHelperClick}
      />
    );

    // Choose how to hide input
    let hide;
    if (isMobile) {
      hide = {
        position: 'absolute',
        visibility: 'hidden',
      };
    } else if (area) {
      hide = layer;
    } else {
      hide = {
        opacity: 0,
        position: 'absolute',
      };
    }

    const inputElement = (
      <input
        {...other}
        checked={checked}
        disabled={disabled}
        name={name}
        ref={(c) => { this.checkbox = c; }}
        style={hide}
        type={inputType}
        value={value}
        onBlur={this.handleBlur}
        onChange={this.handleChange}
        onFocus={this.handleFocus}
      />
    );

    let insertElement = null;
    if (insert) {
      insertElement = React.isValidElement(insert)
        ? insert
        : (
          // eslint-disable-next-line react/no-danger
          <div dangerouslySetInnerHTML={{ __html: insert }} />
        );
    }

    return (
      <div {...wrapProps}>
        {inputElement}
        {insertElement}
        {helper}
      </div>
    );
  }
}

export default Input;
