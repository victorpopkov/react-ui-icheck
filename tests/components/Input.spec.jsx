import React from 'react';
import expect from 'expect';
import { fireEvent, render } from '@testing-library/react';
import Input from '../../src/components/Input';

const testInputComponent = (inputType) => {
  const inputClass = `i${inputType}`;

  describe('when no props are passed', () => {
    it('should match the snapshot', () => {
      const { container } = render(<Input inputType={inputType} />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('when `aria` prop is `true`', () => {
    it(`<div class="${inputClass}" /> should have the corresponding \`aria-checked\` attribute while being clicked`, () => {
      const { container } = render(<Input inputType={inputType} aria />);
      const element = container.querySelector(`.${inputClass}`);
      const inputElement = container.querySelector('input');
      expect(element).toHaveAttribute('aria-checked', 'false');
      fireEvent.click(inputElement);
      expect(element).toHaveAttribute('aria-checked', 'true');
    });
  });

  describe('when `checked` prop is `true`', () => {
    it('<input /> should have "checked" attribute', () => {
      const { container } = render(<Input inputType={inputType} checked />);
      expect(container.querySelector('input')).toHaveAttribute('checked');
    });

    it(`<div class="${inputClass}" /> should have "checked" class`, () => {
      const { container } = render(<Input inputType={inputType} checked />);
      expect(container.querySelector(`.${inputClass}`)).toHaveClass('checked');
    });
  });

  describe('when `className` prop is "test"', () => {
    describe('and `inheritClassName` is `false`', () => {
      it(`<div class="${inputClass}" /> should not have "test" class`, () => {
        const { container } = render(
          <Input
            inputType={inputType}
            className="test"
            inheritClassName={false}
          />,
        );
        expect(container.querySelector(`.${inputClass}`)).not.toHaveClass(
          'test',
        );
      });
    });

    describe('and `inheritClassName` is `true`', () => {
      it(`<div class="${inputClass}" /> should have "test" class`, () => {
        const { container } = render(
          <Input inputType={inputType} className="test" inheritClassName />,
        );
        expect(container.querySelector(`.${inputClass}`)).toHaveClass('test');
      });
    });
  });

  describe('and `disabled` prop is `true`', () => {
    it('<input /> should have "disabled" attribute', () => {
      const { container } = render(<Input inputType={inputType} disabled />);
      const inputElement = container.querySelector('input');
      expect(inputElement).toHaveAttribute('disabled');
      fireEvent.click(inputElement);
      expect(inputElement).toHaveAttribute('disabled');
    });

    it(`<div class="${inputClass}" /> should have "disabled" class`, () => {
      const { container } = render(<Input inputType={inputType} disabled />);
      const element = container.querySelector(`.${inputClass}`);
      expect(element).toHaveClass('disabled');
      fireEvent.click(container.querySelector('input'));
      expect(element).toHaveClass('disabled');
    });
  });

  describe('when `id` prop is "test"', () => {
    describe('and `inheritID` is `false`', () => {
      it(`<div class="${inputClass}" /> should not have "test" class`, () => {
        const { container } = render(
          <Input inputType={inputType} id="test" inheritID={false} />,
        );
        expect(container.querySelector(`.${inputClass}`)).not.toHaveClass(
          'id',
          'iCheck-test',
        );
      });
    });

    describe('and `inheritID` is `true`', () => {
      it(`<div class="${inputClass}" /> should have "test" class`, () => {
        const { container } = render(
          <Input inputType={inputType} id="test" inheritID />,
        );
        expect(container.querySelector(`.${inputClass}`)).toHaveAttribute(
          'id',
          'iCheck-test',
        );
      });
    });
  });

  describe('when `increaseArea` prop', () => {
    const commonStyles = {
      background: 'rgb(255, 255, 255)',
      border: 0,
      cursor: 'pointer',
      display: 'block',
      margin: 0,
      opacity: 0,
      padding: 0,
      position: 'absolute',
    };

    it('is "30%"', () => {
      const { container } = render(
        <Input inputType={inputType} increaseArea="30%" />,
      );
      expect(container.querySelector('input')).toHaveStyle({
        ...commonStyles,
        height: '160%',
        left: '-30%',
        top: '-30%',
        width: '160%',
      });
    });

    it('is "-30%', () => {
      const { container } = render(
        <Input inputType={inputType} increaseArea="-30%" />,
      );
      expect(container.querySelector('input')).toHaveStyle({
        ...commonStyles,
        height: '40%',
        left: '30%',
        top: '30%',
        width: '40%',
      });
    });

    it('is "-60%', () => {
      const { container } = render(
        <Input inputType={inputType} increaseArea="-60%" />,
      );
      expect(container.querySelector('input')).toHaveStyle({
        ...commonStyles,
        height: '0%',
        left: '50%',
        top: '50%',
        width: '0%',
      });
    });
  });

  describe('when `insert` prop is <div>test</div>', () => {
    it('should insert that element after <input />', () => {
      const { container } = render(
        <Input inputType={inputType} insert={<div>test</div>} />,
      );
      const insertElement = container.querySelector('input + div');
      expect(insertElement.tagName).toBe('DIV');
      expect(insertElement.innerHTML).toEqual('test');
    });
  });

  describe('when `isMobile` prop is `true`', () => {
    it('<input /> should have the corresponding style', () => {
      const { container } = render(<Input inputType={inputType} isMobile />);
      expect(container.querySelector('input')).toHaveStyle({
        visibility: 'hidden',
        position: 'absolute',
      });
    });
  });

  describe('when focusing <input />', () => {
    it('should call `onFocus` prop function', () => {
      const onFocus = jest.fn();
      const { container } = render(
        <Input inputType={inputType} onFocus={onFocus} />,
      );
      expect(onFocus).toHaveBeenCalledTimes(0);
      fireEvent.focus(container.querySelector('input'));
      expect(onFocus).toHaveBeenCalledTimes(1);
    });

    it(`<div class="${inputClass}" /> should have "focus" class while being focused`, () => {
      const { container } = render(<Input inputType={inputType} />);
      const element = container.querySelector(`.${inputClass}`);
      const inputElement = container.querySelector('input');
      fireEvent.focus(inputElement);
      expect(element).toHaveClass('focus');
      fireEvent.blur(inputElement);
      expect(element).not.toHaveClass('focus');
    });
  });

  describe('when blurring <input />', () => {
    it('should call `onBlur` prop function', () => {
      const onBlur = jest.fn();
      const { container } = render(
        <Input inputType={inputType} onBlur={onBlur} />,
      );
      expect(onBlur).toHaveBeenCalledTimes(0);
      fireEvent.blur(container.querySelector('input'));
      expect(onBlur).toHaveBeenCalledTimes(1);
    });
  });

  describe('when changing <input />', () => {
    it('should call `onChange` prop function', () => {
      const onChange = jest.fn();
      const { container } = render(
        <Input inputType={inputType} onChange={onChange} />,
      );
      expect(onChange).toHaveBeenCalledTimes(0);
      fireEvent.click(container.querySelector('input'));
      expect(onChange).toHaveBeenCalledTimes(1);
    });
  });

  describe('when changing <ins />', () => {
    describe('and `label` prop is `null`', () => {
      describe('and `insert` prop is `null`', () => {
        it('should not call `onChange` prop function', () => {
          const onChange = jest.fn();
          const { container } = render(
            <Input inputType={inputType} onChange={onChange} />,
          );
          expect(onChange).toHaveBeenCalledTimes(0);
          fireEvent.click(container.querySelector('ins'));
          expect(onChange).toHaveBeenCalledTimes(0);
        });
      });
    });

    describe('and `label` prop is "test"', () => {
      describe('and `insert` prop is "test"', () => {
        describe('and `disabled` prop is `false`', () => {
          it('should call `onChange` prop function', () => {
            const onChange = jest.fn();
            const { container } = render(
              <Input
                inputType={inputType}
                onChange={onChange}
                label="test"
                insert="test"
                disabled={false}
              />,
            );
            expect(onChange).toHaveBeenCalledTimes(0);
            fireEvent.click(container.querySelector('ins'));
            expect(onChange).toHaveBeenCalledTimes(1);
          });
        });

        describe('and `disabled` prop is `true`', () => {
          it('should not call `onChange` prop function', () => {
            const onChange = jest.fn();
            const { container } = render(
              <Input
                inputType={inputType}
                onChange={onChange}
                label="test"
                insert="test"
                disabled
              />,
            );
            expect(onChange).toHaveBeenCalledTimes(0);
            fireEvent.click(container.querySelector('ins'));
            expect(onChange).toHaveBeenCalledTimes(0);
          });
        });
      });
    });
  });
};

describe('<Input />', () => {
  describe('when `inputType` prop is "checkbox"', () => {
    testInputComponent('checkbox');
  });

  describe('when `inputType` prop is "radio"', () => {
    testInputComponent('radio');
  });
});
