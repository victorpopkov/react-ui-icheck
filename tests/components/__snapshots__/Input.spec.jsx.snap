// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` and the \`aria\` prop is \`true\` as shallow should match the snapshot 1`] = `
<div
  aria-checked={false}
  className="icheckbox"
  role="checkbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "100%",
        "left": "0%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "0%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` and the \`increaseArea\` prop is \`-30%\` as shallow should match the snapshot 1`] = `
<div
  className="icheckbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "40%",
        "left": "30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "30%",
        "width": "40%",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "40%",
        "left": "30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "30%",
        "width": "40%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` and the \`increaseArea\` prop is \`-60%\` as shallow should match the snapshot 1`] = `
<div
  className="icheckbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` and the \`increaseArea\` prop is \`30%\` as shallow should match the snapshot 1`] = `
<div
  className="icheckbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` and the \`insert\` prop is \`<div>test</div>} />\` as render should match the snapshot 1`] = `
<div
  class="icheckbox"
>
  <input
    style="opacity:0;position:absolute"
    type="checkbox"
    value=""
  />
  <div>
    test
  </div>
  <ins
    class="iCheck-helper"
    role="presentation"
    style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` and the \`isMobile\` prop is \`true\` as shallow an \`<input />\` should match style attribute to the snapshot 1`] = `
Object {
  "position": "absolute",
  "visibility": "hidden",
}
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` as mount should match the default props snapshot 1`] = `
Object {
  "activeClassName": null,
  "aria": false,
  "checkboxClassName": "icheckbox",
  "checked": false,
  "checkedClassName": "checked",
  "className": null,
  "disabled": false,
  "disabledClassName": "disabled",
  "focusClassName": "focus",
  "hoverClassName": "hover",
  "hovered": false,
  "id": null,
  "increaseArea": "",
  "indeterminate": false,
  "indeterminateClassName": "indeterminate",
  "inheritClassName": false,
  "inheritID": false,
  "inputType": "checkbox",
  "insert": null,
  "isMobile": false,
  "label": null,
  "name": null,
  "onBlur": null,
  "onChange": null,
  "onFocus": null,
  "radioClassName": "iradio",
  "value": "",
}
`;

exports[`Input component when the required \`inputType\` prop is set to \`checkbox\` as mount should match the default state snapshot 1`] = `
Object {
  "active": false,
  "checked": false,
  "focused": false,
  "hovered": false,
  "indeterminate": false,
  "prevPropChecked": false,
}
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` and the \`aria\` prop is \`true\` as shallow should match the snapshot 1`] = `
<div
  aria-checked={false}
  className="iradio"
  role="radio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "100%",
        "left": "0%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "0%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` and the \`increaseArea\` prop is \`-30%\` as shallow should match the snapshot 1`] = `
<div
  className="iradio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "40%",
        "left": "30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "30%",
        "width": "40%",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "40%",
        "left": "30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "30%",
        "width": "40%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` and the \`increaseArea\` prop is \`-60%\` as shallow should match the snapshot 1`] = `
<div
  className="iradio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` and the \`increaseArea\` prop is \`30%\` as shallow should match the snapshot 1`] = `
<div
  className="iradio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` and the \`insert\` prop is \`<div>test</div>} />\` as render should match the snapshot 1`] = `
<div
  class="iradio"
>
  <input
    style="opacity:0;position:absolute"
    type="radio"
    value=""
  />
  <div>
    test
  </div>
  <ins
    class="iCheck-helper"
    role="presentation"
    style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
  />
</div>
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` and the \`isMobile\` prop is \`true\` as shallow an \`<input />\` should match style attribute to the snapshot 1`] = `
Object {
  "position": "absolute",
  "visibility": "hidden",
}
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` as mount should match the default props snapshot 1`] = `
Object {
  "activeClassName": null,
  "aria": false,
  "checkboxClassName": "icheckbox",
  "checked": false,
  "checkedClassName": "checked",
  "className": null,
  "disabled": false,
  "disabledClassName": "disabled",
  "focusClassName": "focus",
  "hoverClassName": "hover",
  "hovered": false,
  "id": null,
  "increaseArea": "",
  "indeterminate": false,
  "indeterminateClassName": "indeterminate",
  "inheritClassName": false,
  "inheritID": false,
  "inputType": "radio",
  "insert": null,
  "isMobile": false,
  "label": null,
  "name": null,
  "onBlur": null,
  "onChange": null,
  "onFocus": null,
  "radioClassName": "iradio",
  "value": "",
}
`;

exports[`Input component when the required \`inputType\` prop is set to \`radio\` as mount should match the default state snapshot 1`] = `
Object {
  "active": false,
  "checked": false,
  "focused": false,
  "hovered": false,
  "indeterminate": false,
  "prevPropChecked": false,
}
`;
